File: androidApp/src/main/java/com/aitrainingapp/android/ui/training_history/TrainingHistoryScreen.kt
package com.aitrainingapp.android.ui.training_history

import androidx.compose.foundation.layout.*
import androidx.compose.foundation.lazy.LazyColumn
import androidx.compose.material.icons.Icons
import androidx.compose.material.icons.filled.ArrowDropDown
import androidx.compose.material.icons.filled.ArrowDropUp
import androidx.compose.material3.*
import androidx.compose.runtime.*
import androidx.compose.ui.Alignment
import androidx.compose.ui.Modifier
import androidx.compose.ui.graphics.Color
import androidx.compose.ui.unit.dp
import com.aitrainingapp.android.viewmodel.TrainingHistoryViewModel
import com.aitrainingapp.util.formatDate

@Composable
fun TrainingHistoryScreen(viewModel: TrainingHistoryViewModel) {
    val history by viewModel.history.collectAsState()

    LaunchedEffect(Unit) {
        viewModel.loadHistory()
    }

    Surface(
        modifier = Modifier.fillMaxSize(),
        color = MaterialTheme.colorScheme.background
    ) {
        Column(modifier = Modifier.padding(16.dp)) {
            Spacer(modifier = Modifier.height(20.dp))

            Text(
                text = "Historia treningów",
                style = MaterialTheme.typography.headlineMedium,
                color = MaterialTheme.colorScheme.onBackground
            )

            Spacer(modifier = Modifier.height(16.dp))

            LazyColumn(verticalArrangement = Arrangement.spacedBy(8.dp)) {
                val reversedHistory = history.reversed()
                items(reversedHistory.size) { index ->
                    val item = reversedHistory[index]
                    val volume = item.weight * item.reps * item.sets
                    val previousVolume = if (index < reversedHistory.lastIndex) {
                        val prev = reversedHistory[index + 1]
                        prev.weight * prev.reps * prev.sets
                    } else volume

                    val delta = volume - previousVolume
                    val (arrow, deltaColor) = when {
                        delta > 0 -> Icons.Default.ArrowDropUp to Color.Green
                        delta < 0 -> Icons.Default.ArrowDropDown to Color.Red
                        else -> null to Color.Gray
                    }

                    Card(
                        modifier = Modifier.fillMaxWidth(),
                        colors = CardDefaults.cardColors(containerColor = MaterialTheme.colorScheme.surfaceVariant)
                    ) {
                        Row(
                            modifier = Modifier
                                .fillMaxWidth()
                                .padding(16.dp),
                            verticalAlignment = Alignment.CenterVertically
                        ) {
                            Column(
                                modifier = Modifier.width(80.dp),
                                horizontalAlignment = Alignment.CenterHorizontally
                            ) {
                                Text(
                                    text = "${volume.toInt()} kg",
                                    color = MaterialTheme.colorScheme.onSurface,
                                    style = MaterialTheme.typography.titleSmall
                                )
                                arrow?.let {
                                    Icon(
                                        imageVector = it,
                                        contentDescription = null,
                                        tint = deltaColor,
                                        modifier = Modifier.size(32.dp)
                                    )
                                    Text(
                                        text = "${kotlin.math.abs(delta).toInt()}",
                                        color = deltaColor,
                                        fontSize = MaterialTheme.typography.titleMedium.fontSize
                                    )
                                }
                            }

                            Spacer(modifier = Modifier.width(16.dp))

                            Column {
                                Text("Typ: ${item.trainingType ?: "Nieznany"}", color = MaterialTheme.colorScheme.onSurface)
                                Text("Waga: ${item.weight} kg", color = MaterialTheme.colorScheme.onSurface)
                                Text("Powtórzenia: ${item.reps}", color = MaterialTheme.colorScheme.onSurface)
                                Text("Serie: ${item.sets}", color = MaterialTheme.colorScheme.onSurface)
                                Text("RPE: ${item.rpe}", color = MaterialTheme.colorScheme.onSurface)
                                Text("Data: ${formatDate(item.dateTime)}", color = MaterialTheme.colorScheme.onSurface)
                            }
                        }
                    }
                }
            }
        }
    }
}

File: iosApp/iosApp/ui/TrainingHistoryScreen.swift
import SwiftUI
import Shared

struct TrainingHistoryScreen: View {
    @ObservedObject var viewModel: TrainingHistoryViewModelWrapper

    var body: some View {
        ScrollView {
            VStack(alignment: .leading, spacing: 16) {
                Text("Historia treningów")
                    .font(.title)
                    .bold()

                ForEach(viewModel.history.reversed().indices, id: \.self) { index in
                    let item = viewModel.history.reversed()[index]
                    let volume = item.weight * Double(item.reps * item.sets)

                    let previousVolume: Double = {
                        if index < viewModel.history.count - 1 {
                            let prev = viewModel.history.reversed()[index + 1]
                            return prev.weight * Double(prev.reps * prev.sets)
                        } else {
                            return volume
                        }
                    }()

                    let delta = volume - previousVolume
                    let arrowIcon: String? = delta > 0 ? "arrow.up" :
                                             delta < 0 ? "arrow.down" : nil
                    let deltaColor: Color = delta > 0 ? .green :
                                            delta < 0 ? .red : .gray

                    VStack(alignment: .leading, spacing: 8) {
                        HStack {
                            VStack {
                                Text("\(Int(volume)) kg")
                                    .font(.headline)
                                if let arrow = arrowIcon {
                                    Image(systemName: arrow)
                                        .foregroundColor(deltaColor)
                                    Text("\(abs(Int(delta)))")
                                        .foregroundColor(deltaColor)
                                        .font(.subheadline)
                                }
                            }
                            .frame(width: 80)

                            VStack(alignment: .leading, spacing: 4) {
                                Text("Typ: \(item.trainingType ?? "Nieznany")")
                                Text("Waga: \(item.weight, specifier: "%.1f") kg")
                                Text("Powtórzenia: \(item.reps)")
                                Text("Serie: \(item.sets)")
                                Text("RPE: \(item.rpe)")
                                Text("Data: \(formatDate(item.dateTime))")
                            }
                            .font(.subheadline)
                        }
                        .padding()
                        .background(Color(UIColor.secondarySystemBackground))
                        .cornerRadius(10)
                    }
                }
            }
            .padding()
        }
        .onAppear {
            viewModel.load()
        }
    }
}
